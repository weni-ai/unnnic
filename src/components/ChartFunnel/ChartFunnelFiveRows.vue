<template>
  <ChartFunnelBaseRow
    class="unnnic-chart-funnel-five-rows"
    :rows="rows"
    viewBox="0 0 361 220"
  />
</template>

<script>
import ChartFunnelBaseRow from './ChartFunnelBaseRow.vue';
export default {
  name: 'UnnnicChartFunnelFiveRows',

  components: {
    ChartFunnelBaseRow,
  },

  props: {
    data: {
      type: Array,
      required: true,
    },
  },

  computed: {
    rows() {
      const { data } = this;
      return [
        {
          pathD:
            'M352.506 0H8.42251C1.93921 0 -1.85142 7.30766 1.88266 12.6076L24 44H336.75L359.028 12.6324C362.79 7.33528 359.003 0 352.506 0Z',
          titleX: '50%',
          titleY: '20',
          descriptionX: '50%',
          descriptionY: '38',
          title: data[0].title,
          description: data[0].description,
        },
        {
          pathD: 'M337 44H24L56.4654 88H304.535L337 44Z',
          titleX: '50%',
          titleY: '62',
          descriptionX: '50%',
          descriptionY: '80',
          title: data[1].title,
          description: data[1].description,
        },
        {
          pathD: 'M304.5 88H56.5L88.9564 132H273.708L304.5 88Z',
          titleX: '50%',
          titleY: '106',
          descriptionX: '50%',
          descriptionY: '124',
          title: data[2].title,
          description: data[2].description,
        },
        {
          pathD: 'M274 132H89L121 176H242L274 132Z',
          titleX: '50%',
          titleY: '150',
          descriptionX: '50%',
          descriptionY: '168',
          title: data[3].title,
          description: data[3].description,
        },
        {
          pathD:
            'M242 176H120.957L130.481 213.947C131.374 217.505 134.573 220 138.241 220H223.89C227.498 220 230.659 217.585 231.608 214.105L242 176Z',
          titleX: '50%',
          titleY: '194',
          descriptionX: '50%',
          descriptionY: '212',
          title: data[4].title,
          description: data[4].description,
        },
      ];
    },
  },
};
</script>

<style lang="scss">
@import '../../assets/scss/unnnic.scss';

.unnnic-chart-funnel-five-rows {
  [class$='row']:nth-child(1) {
    fill: $unnnic-color-weni-950;
  }
  [class$='row']:nth-child(2) {
    fill: $unnnic-color-weni-900;
  }
  [class$='row']:nth-child(3) {
    fill: $unnnic-color-weni-800;
  }
  [class$='row']:nth-child(4) {
    fill: $unnnic-color-weni-700;
  }
  [class$='row']:nth-child(5) {
    fill: $unnnic-color-weni-600;
  }
}
</style>
