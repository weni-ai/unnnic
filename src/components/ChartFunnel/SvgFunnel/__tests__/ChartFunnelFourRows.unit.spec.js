import { mount } from '@vue/test-utils';
import { beforeEach, describe, expect, afterEach, test } from 'vitest';
import ChartFunnelFiveRows from '@/components/ChartFunnel/SvgFunnel/ChartFunnelFourRows.vue';

describe('ChartFunnelFourRows.vue', () => {
  let wrapper;
  const mockData = [
    { title: 'Row 1', description: 'Description 1' },
    { title: 'Row 2', description: 'Description 2' },
    { title: 'Row 3', description: 'Description 3' },
    { title: 'Row 4', description: 'Description 4' },
  ];

  beforeEach(() => {
    wrapper = mount(ChartFunnelFiveRows, {
      props: {
        data: mockData,
      },
      global: {
        stubs: {
          ChartFunnelBaseRow: {
            template: '<div class="chart-funnel-base-row" />',
            props: ['rows', 'viewBox'],
          },
        },
      },
    });
  });

  afterEach(() => {
    wrapper.unmount();
  });

  test('renders correctly', () => {
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.classes()).toContain('unnnic-chart-funnel-four-rows');
  });

  test('renders ChartFunnelBaseRow component with correct props', () => {
    const baseRow = wrapper.findComponent('.unnnic-chart-funnel-four-rows');

    expect(baseRow.exists()).toBe(true);
    expect(baseRow.props('viewBox')).toBe('0 0 365 224');
  });

  test('computes rows correctly with provided data', () => {
    const computedRows = wrapper.vm.rows;

    expect(computedRows).toHaveLength(4);

    expect(computedRows[0]).toEqual({
      pathD:
        'M356.374 0H8.57255C2.47804 0 -1.37827 6.54249 1.5734 11.8745L26 56H338.75L363.36 11.8984C366.336 6.56589 362.481 0 356.374 0Z',
      titleX: '50%',
      titleY: '23',
      descriptionX: '50%',
      descriptionY: '43',
      title: mockData[0].title,
      description: mockData[0].description,
    });

    expect(computedRows[1]).toEqual({
      pathD: 'M339 56H26L58.4654 112H306.535L339 56Z',
      titleX: '50%',
      titleY: '78',
      descriptionX: '50%',
      descriptionY: '98',
      title: mockData[1].title,
      description: mockData[1].description,
    });

    expect(computedRows[2]).toEqual({
      pathD: 'M306.5 112H58.5L90.9564 168H275.708L306.5 112Z',
      titleX: '50%',
      titleY: '136',
      descriptionX: '50%',
      descriptionY: '156',
      title: mockData[2].title,
      description: mockData[2].description,
    });
  });

  test('updates rows when data prop changes', async () => {
    const newData = [
      { title: 'New 1', description: 'New Desc 1' },
      { title: 'New 2', description: 'New Desc 2' },
      { title: 'New 3', description: 'New Desc 3' },
      { title: 'New 4', description: 'New Desc 4' },
    ];

    await wrapper.setProps({ data: newData });
    const computedRows = wrapper.vm.rows;

    expect(computedRows[0].title).toBe('New 1');
    expect(computedRows[0].description).toBe('New Desc 1');
    expect(computedRows[3].title).toBe('New 4');
    expect(computedRows[3].description).toBe('New Desc 4');
  });

  test('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('passes correct viewBox to ChartFunnelBaseRow', () => {
    const baseRow = wrapper.findComponent('.unnnic-chart-funnel-four-rows');
    expect(baseRow.props('viewBox')).toBe('0 0 365 224');
  });

  test('correctly formats all row paths', () => {
    const computedRows = wrapper.vm.rows;
    const expectedPaths = [
      'M356.374 0H8.57255C2.47804 0 -1.37827 6.54249 1.5734 11.8745L26 56H338.75L363.36 11.8984C366.336 6.56589 362.481 0 356.374 0Z',
      'M339 56H26L58.4654 112H306.535L339 56Z',
      'M306.5 112H58.5L90.9564 168H275.708L306.5 112Z',
      'M276 168H91L120.697 219.969C122.121 222.462 124.772 224 127.643 224H239.357C242.228 224 244.879 222.462 246.303 219.969L276 168Z',
    ];

    computedRows.forEach((row, index) => {
      expect(row.pathD).toBe(expectedPaths[index]);
    });
  });
});
