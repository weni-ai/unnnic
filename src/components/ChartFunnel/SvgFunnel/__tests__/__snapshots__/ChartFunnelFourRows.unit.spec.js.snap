// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChartFunnelFourRows.vue > matches snapshot 1`] = `"<div class="chart-funnel-base-row unnnic-chart-funnel-four-rows"></div>"`;
